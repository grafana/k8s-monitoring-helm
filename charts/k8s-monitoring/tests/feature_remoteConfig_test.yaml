# yamllint disable rule:document-start rule:line-length rule:trailing-spaces
suite: Feature - Remote Config
templates:
  - alloy-config.yaml
tests:
  - it: renders the remotecfg component with the right default attributes
    template: alloy-config.yaml
    set:
      cluster:
        name: ci-test-cluster
      alloy-metrics:
        enabled: true
        remoteConfig:
          enabled: true
          url: https://fleet-management-prod-001.grafana.net
          auth:
            type: basic
            username: 123456
            password: "It's a secret to everyone"
    asserts:
      - isKind:
          of: ConfigMap
      - matchSnapshot:
          path: data["config.alloy"]

  - it: allows for overriding the default attributes
    template: alloy-config.yaml
    set:
      cluster:
        name: ci-test-cluster
      alloy-metrics:
        enabled: true
        remoteConfig:
          enabled: true
          url: https://fleet-management-prod-001.grafana.net
          auth:
            type: basic
            username: 123456
            password: "It's a secret to everyone"
          extraAttributes:
            color: blue
            release: stable  # overrides the default release value
            sourceVersion: null  # removes the sourceVersion attribute
    asserts:
      - isKind:
          of: ConfigMap
      - matchSnapshot:
          path: data["config.alloy"]
