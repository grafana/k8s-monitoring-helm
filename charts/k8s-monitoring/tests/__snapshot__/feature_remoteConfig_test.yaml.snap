allows for overriding the default attributes:
  1: |
    |
      remote.kubernetes.secret "alloy_metrics_remote_cfg" {
        name      = "alloy-metrics-remote-cfg-release-name-k8s-monitoring"
        namespace = "NAMESPACE"
      }

      remotecfg {
        id = sys.env("GCLOUD_FM_COLLECTOR_ID")
        url = "https://fleet-management-prod-001.grafana.net"
        basic_auth {
          username = convert.nonsensitive(remote.kubernetes.secret.alloy_metrics_remote_cfg.data["username"])
          password = remote.kubernetes.secret.alloy_metrics_remote_cfg.data["password"]
        }
        poll_frequency = "5m"
        attributes = {
          "cluster" = "ci-test-cluster",
          "color" = "blue",
          "namespace" = "NAMESPACE",
          "platform" = "kubernetes",
          "release" = "stable",
          "source" = "k8s-monitoring",
          "workloadName" = "alloy-metrics",
          "workloadType" = "statefulset",
        }
      }
renders the remotecfg component with the right default attributes:
  1: |
    |
      remote.kubernetes.secret "alloy_metrics_remote_cfg" {
        name      = "alloy-metrics-remote-cfg-release-name-k8s-monitoring"
        namespace = "NAMESPACE"
      }

      remotecfg {
        id = sys.env("GCLOUD_FM_COLLECTOR_ID")
        url = "https://fleet-management-prod-001.grafana.net"
        basic_auth {
          username = convert.nonsensitive(remote.kubernetes.secret.alloy_metrics_remote_cfg.data["username"])
          password = remote.kubernetes.secret.alloy_metrics_remote_cfg.data["password"]
        }
        poll_frequency = "5m"
        attributes = {
          "cluster" = "ci-test-cluster",
          "namespace" = "NAMESPACE",
          "platform" = "kubernetes",
          "release" = "RELEASE-NAME",
          "source" = "k8s-monitoring",
          "sourceVersion" = "3.4.0",
          "workloadName" = "alloy-metrics",
          "workloadType" = "statefulset",
        }
      }
